src/frontends/onnx/frontend/src/op/dft.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/dft.hpp"
+
+#include "default_opset.hpp"
+#include "utils/common.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+
+namespace {
+// For DFT, IDFT, IRDFT cases, if real data are provided (with shape [D_0, D_1, ..., D_{N-1}, 1])
+// it's needed to fill tensors with zero imaginary part to be aligned with Core ops requirements.
+ov::Output<ov::Node> convert_real_to_complex_if_needed(const ov::Output<ov::Node>& data,
```

Comment:
@mateusztabaka Do you think that it makes sense to fill with zero imaginary part also IDFT, IRDFT case? ONNX spec is not very clear for such cases (in ONNX tests there is only case of using DFT(inversed=0,oneside=0) with real numbers as input)

Reply:
> Do you think that it makes sense to fill with zero imaginary part also IDFT, IRDFT case

Yes, I think so.

Also, should we pad output of IRDFT with zeros?

Reply:
@mateusztabaka I've checked Pytorch and I don't see unsqueeze for IRDFT output case ([..,..,1]), for ONNXRT IRDFT (inverse + onesided) is not supported